### PR TITLE
Django3 upgrade

### PR DESCRIPTION
**Pull request by aleksandr-shtaub**
- Drop support for python<3.7 and remove old python 2 code
- Drop support for django<2.2
- Use safe_load() instead of load() from pyyaml=6.0
- Update CI configuration

### CODE REVIEW

**Comment by aleksandr-shtaub** [comment]
TODO in next PRs:

- install `cached-property` only when "python<3.8"
- add dev requirements.txt

**Review by toopy — APPROVED** [review]
:1st_place_medal:

**Comment by ewjoachim** [comment]
For the future: please try to put the manual changes and the automated changes (e.g. black, pyupgrade etc) in different commits. It helps when reviewing :)

File: .travis.yml [hunk 1]
```diff
@@ -2,18 +2,20 @@ language: python
...
+    - python: 3.8
+      env: TOX_ENV=py38-django22
+    - python: 3.8
+      env: TOX_ENV=py38-django32
+    - python: 3.9
+      env: TOX_ENV=py39-django22
+    - python: 3.9
+      env: TOX_ENV=py39-django32
+    - python: 3.9
+      env: TOX_ENV=py39-djangostable
```
Comment (ewjoachim) [inline .travis.yml:30]: There are things that work with Django 2.2, and 3.2 but not 3.0 (which is deprecated, I know). I'm not saying you HAVE to support 3.0 and 3.1, but just, be aware that you may run into surprises while upgrading, given that the only supported path for upgrading is to go through all intermediate versions.

File: tests/django/test_command.py [hunk 2]
```diff
@@ -142,11 +142,12 @@ def test_call_command(capsys):
...
       generator: Choices
-      choices: [1, 2]
-""".format(
-        username_length=30 if django.VERSION < (1, 10) else 150,
-        lastname_length=30 if django.VERSION < (2,) else 150,
-    )
+      choices:
+      - 1
+      - 2
+"""
```
Comment (ewjoachim) [inline tests/django/test_command.py]: Files should always end with a new line. You can set your editor to take care of that.

**Review by ewjoachim — APPROVED** [review]

**Review by toopy — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (aleksandr-shtaub) [reply .travis.yml:30]: Yeah, you're right. I've added intermediate versions to the test suite, just to be sure we won't get any issues with them.

File: tests/django/test_command.py (hunk 2 above)
Comment (aleksandr-shtaub) [reply tests/django/test_command.py]: I'm still on the process of migration from pycharm to vscode. I've upgraded my configs, thanks.

**Pull request merged**